views-core/src/main/java/io/micronaut/views/model/security/CsrfViewModelProcessor.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.views.model.security;
+
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.security.csrf.repository.CsrfTokenRepository;
+import io.micronaut.views.model.MapViewModelProcessor;
+import jakarta.inject.Singleton;
+
+import java.util.Map;
+
+/**
+ * Adds a CSRF Token to the model map if a CSRF Token can be retrieved via {@link CsrfTokenRepository}.
+ *
+ * @author Sergio del Amo
+ * @since 5.6.0
+ */
+@Requires(property = SecurityViewModelProcessorConfigurationProperties.PREFIX + ".enabled", notEquals = StringUtils.FALSE)
+@Requires(beans = {CsrfTokenRepository.class, CsrfViewModelProcessorConfiguration.class})
+@Requires(classes = HttpRequest.class)
+@Singleton
+public class CsrfViewModelProcessor extends MapViewModelProcessor {
```

Comment:
should this be `@Internal` given its parent is? Should it be `final`?

Reply:
resolved via https://github.com/micronaut-projects/micronaut-views/pull/914/commits/2d11589961628cd66afc158cb83b976b14009a27